src/GslibIO.jl

```diff
@@ -155,4 +198,53 @@ function save(file::File{format"GSLIB"}, sdata::AbstractData)
        propnames=vars)
 end
 
+# low level function for saving data to a legacy GSLIB format
+function save_legacy(filename::AbstractString, data::AbstractMatrix,
+                     varnames::NTuple, header::AbstractString, na)
+  @assert size(data, 2) == length(varnames) "Invalid data for the specified variable names"
+  nvars = size(data, 2)
+
+  # handle missing values
+  datana = replace!(data, NaN=>na)
+
+  open(filename, "w") do f
+    write(f, "$header\n")
+    write(f, "$nvars\n")
+    for v in varnames
+      write(f, "$v\n")
+    end
+    writedlm(f, datana, ' ')
+  end
+end
+
+"""
+    save_legacy(filename, sdata)
+
+Save spatial data `sdata` to `filename` in legacy GSLIB format. Optionally, specify
+the `coordnames`, the `header` and the value `na` used to represent missing entries.
+"""
+function save_legacy(filename::AbstractString, sdata::AbstractData; coordnames=(:x, :y, :z),
+                     header="# This file was generated with GslibIO.jl", na=-999.0)
+  table = values(sdata)
+  sdomain = domain(sdata)
+  
+  if sdomain isa PointSet
+    # add coordinates to data and coordnames to varnames
+    coords = coordinates(sdomain)
+    cdim = ncoords(sdomain)
+    @assert cdim == length(coordnames) "The length of coordinate names must be equal to the coordinate dimension"
```

Comment:
```suggestion
    @assert ncoords(sdomain) == length(coordnames) "The length of coordinate names must be equal to the coordinate dimension"
```
Try to not introduce variables when they are not used in multiple places later.